tests/foreman/virtwho/cli/test_esx_sca.py

```diff
@@ -502,3 +505,95 @@ def test_positive_remove_env_option(
+
+        # verify the two service accounts belonging to the same org are the same
+        assert rhsm_username[0] == rhsm_username[1]
+
+        # Create a different org virtwho_fake_XXXX and then create virt-who config in that org, get the service account rhsm_username
+        ORG_DATA = {'name': f'virtwho_fake_{gen_string("alpha")}'}
+        org = target_sat.api.Organization(name=ORG_DATA['name']).create()
+        target_sat.api.Location(organization=[org]).create()
+        form_data_cli['organization-id'] = org.id
+        form_data_cli['name'] = gen_string('alpha')
+        virtwho_config_cli = target_sat.cli.VirtWhoConfig.create(form_data_cli)[
+            'general-information'
+        ]
+        command = get_configure_command(virtwho_config_cli['id'], org.name)
+        deploy_configure_by_command(
+            command, form_data_cli['hypervisor-type'], debug=True, org=org.label
+        )
+        config_file = get_configure_file(virtwho_config_cli['id'])
+        rhsm_username.append(get_configure_option('rhsm_username', config_file))
+        vc_id.append(virtwho_config_cli['id'])
+
+        # verify the two service accounts belonging to different orgs are different
+        assert rhsm_username[2] != rhsm_username[0]
+
+        # Verify rhsm.log contains correct orgs to which hypervisor-guest mapping is sent to, and rhsm.log contains the correct user names.
+        runcmd(get_configure_command(vc_id[0], module_sca_manifest_org.name))
+        runcmd(get_configure_command(vc_id[1], module_sca_manifest_org.name))
+        service_account_message = [
+            f"Authenticating with RHSM username {rhsm_username[0]}",
+            f"Authenticating with RHSM username {rhsm_username[2]}",
+            f"Host-to-guest mapping being sent to '{module_sca_manifest_org.name}'",
+            f"Host-to-guest mapping being sent to '{org.name}'",
+        ]
+        for item in service_account_message:
+            assert check_message_in_rhsm_log(item)
+
+        # delete one of the virt-who configs belonging to module_sca_manifest_org, verify the other service account for that org still exists
+        # delete the other config in that org, verify the service account doesn't exist anymore
+        vd_ids = [vc_id[0], vc_id[1]]
+        messages = [
+            f"Authenticating with RHSM username {rhsm_username[0]}",
+            f"Authenticating with RHSM username {rhsm_username[1]}",
+        ]
+        for index, id in enumerate(vd_ids):
+            target_sat.cli.VirtWhoConfig.delete({'id': id})
+            config_file = get_configure_file(id)
+            runcmd(f"rm -f {config_file}")
+            if index == 0:
+                runcmd(get_configure_command(id, module_sca_manifest_org.name))
+                assert check_message_in_rhsm_log(messages[0])
+            elif index == 1:
+                restart_virtwho_service()
+                assert not check_message_in_rhsm_log(messages[1])
```

Comment:
Maybe I'm missing something as I'm not SME but why not:

```
runcmd(get_configure_command(id, module_sca_manifest_org.name))
restart_virtwho_service()
assert check_message_in_rhsm_log(messages[index]) == (index == 0)
  
```

Reply:
Is it because with index == 1, the deploy command would fail?

Reply:
@lhellebr when delete all the virt-who config belong to module_sca_manifest_org, verify service account  rhsm_username_1 does not exist.

Reply:
@lhellebr Hello Lukas, I have fixed all the suggestions, could you please review it again? Thank you very much

Reply:
@yanpliu Hello, you already have my ACK